test/salt/pillar/top.sls

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# vim: ft=yaml
+---
+base:
+  '*':
+    - TEMPLATE
+    - define_roles
+  'os_family:Gentoo':
+    - match: grain
+    - gentoo
```

Comment:
While this doesn't affect this specific PR, there's another slight issue to consider here.

The ordering of the entries is significant, since the pillars are merged in the order given.  So we'll now be merging the `gentoo` pillar at the very end, rather than between `TEMPLATE` and `define_roles`.  What will we do about situations where we need the platform-specific pillar merged in the middle?

Of course, this applies whether we keep this file or keep this in `kitchen.yml` as mentioned in the previous comment.

Reply:
In some situation I need to apply some pillar at the very last for all minions, I do something like the following:

```yaml
# -*- coding: utf-8 -*-
# vim: ft=yaml
---
base:
  '*':
    - TEMPLATE
  'os_family:Gentoo':
    - match: grain
    - gentoo
  '*':
    - define_roles
...
```

Reply:
@baby-gnu So that works?  That's actually a `yamllint` violation ([`key-duplicates`](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.key_duplicates)).  I suppose we could use a `yamllint` ignore for those lines, if we ever need them.

Reply:
Ho yes, my bad, I thought I did that but in fact I cheated with:

```yaml
base:
  '*':
    - global_vars
  'domain:example.net':
    - match: grain
    - example_net
# […]
  'os:*':
    - match: grain
    - mine
```